From 9ec332b67605f275d8a29699cb542676ef99ed23 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 20 Jan 2015 14:28:15 +0000
Subject: [PATCH 1/2] Win32: Define WIN32_LEAN_AND_MEAN in config.h

.. to avoid 'interface' conflict between OLE (which defines it as 'struct')
and libusb-compat which has 'interface' as a struct member.

replacements.h did define this already, but it was too late since
helper/system.h and helper/types.h are included before replcaements.h
and they end up pulling in windows.h
---
 configure.ac              | 4 ++++
 src/helper/replacements.h | 1 -
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ce28633..eea9664 100644
--- a/configure.ac
+++ b/configure.ac
@@ -804,6 +804,10 @@ then
   AC_MSG_ERROR([The option: with_ftd2xx_linux_tardir is for LINUX only.])
 fi
 
+if test $is_win32 = yes; then
+  AC_DEFINE([WIN32_LEAN_AND_MEAN], [1], [1 to avoid ole 'interface' C macro conflict w/ libusb-compat])
+fi
+
 if test $build_ft2232_ftd2xx = yes -o $build_presto_ftd2xx = yes -o $build_usb_blaster_ftd2xx = yes -o $build_openjtag_ftd2xx = yes; then
   AC_MSG_CHECKING([for ftd2xx.lib exists (win32)])
 
diff --git a/src/helper/replacements.h b/src/helper/replacements.h
index 565fe9d..2776602 100644
--- a/src/helper/replacements.h
+++ b/src/helper/replacements.h
@@ -135,7 +135,6 @@ static inline unsigned usleep(unsigned int usecs)
 /* Windows specific */
 #ifdef _WIN32
 
-#define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 #include <time.h>
 
-- 
2.2.2

